### PR TITLE
List all approved premises

### PR DESCRIPTION
**Pull request by pezholio**
This is a simple end to end journey that fetches premises from the API, and presents them in a table view.

We have a client that fetches the data, a service that calls the client and marshals the data into the correct shape, and then a controller that calls the service.

This should be enough to demonstrate a pattern we can replicate across the app. There's definitely room for improvement / further optimisations along the way!

# Screenshot

image

### CODE REVIEW

**Review by richpjames — APPROVED** [review]
Wonderful! 
I think the approach/structure should serve us really well

File: server/services/index.ts [hunk 1]
```diff
@@ -1,16 +1,20 @@
 import { dataAccess } from '../data'
+
 import UserService from './userService'
+import PremisesService from './premisesService'
 
 export const services = () => {
-  const { hmppsAuthClient } = dataAccess()
+  const data = dataAccess()
```
[the remark this answers is not in the document]
Comment (pezholio) [reply server/services/index.ts:7]: I tried that, but I got an ESlint error, but that seems to have gone now!

[the remark this answers is not in the document]
Comment (richpjames) [reply server/services/index.ts:7]: oh yeah I've come across that before in alpha - I don't really understand the rationale behind that rule

File: integration_tests/.eslintrc [hunk 2]
```diff
@@ -7,5 +7,12 @@
   "rules": {
     "import/no-extraneous-dependencies": ["error", {"devDependencies": true}],
     "no-only-tests/no-only-tests": "error"
+  },
+  "settings": {
+    "import/resolver": {
+      "typescript": {
+        "alwaysTryTypes": true
+      }
+    }
```
[the remark this answers is not in the document]
Comment (pezholio) [reply integration_tests/.eslintrc:16]: I've kept the typescript object, but `alwaysTryTypes` is no longer needed

File: integration_tests/tsconfig.json [hunk 3]
```diff
@@ -3,8 +3,9 @@
     "target": "es5",
     "noEmit": true,
     "lib": ["es5", "dom", "es2015.promise"],
-    "types": ["cypress"],
-    "esModuleInterop": true
+    "types": ["cypress", "approved-premises"],
+    "esModuleInterop": true,
+    "typeRoots": ["../server/@types", "../node_modules/@types"]
```
[the remark this answers is not in the document]
Comment (pezholio) [reply integration_tests/tsconfig.json:8]: Yeah, this is all still relevant 👍

File: server/data/premisesClient.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import type { Premises } from 'approved-premises'
+import RestClient from './restClient'
+import config, { ApiConfig } from '../config'
+
+export default class PremisesClient {
+  restClient: RestClient
+
+  constructor(token: string) {
+    this.restClient = new RestClient('welcomeClient', config.apis.approvedPremises as ApiConfig, token)
```
[the remark this answers is not in the document]
Comment (pezholio) [reply server/data/premisesClient.ts:9]: Good catch 👍

**Pull request merged**